### PR TITLE
feat(xds): Add support nomad service discovery to the ScrapeConfig CRD

### PR DESCRIPTION
**Pull request by dongjiang1989**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

Add support nomad service discovery to the ScrapeConfig CRD

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
feat(xds): Add support nomad service discovery to the ScrapeConfig CRD
```

### CODE REVIEW

**Comment by dongjiang1989** [comment]
@slashpai  PTAL

**Comment by nicolastakashi** [comment]
Thanks for handling that @dongjiang1989, LGTM 🥳 
I'm going to approve but waiting for the last review from @slashpai since she's more aware of other implementations.

**Review by nicolastakashi — DISMISSED** [review]

File: pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go
Comment (slashpai) [inline pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go:871]: nit: lets remove this line

File: pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go
Comment (slashpai) [inline pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go:873]: nit: missed `omitempty` for optional field

File: pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go
Comment (slashpai) [inline pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go:875]: missed `omitempty` for optional field

File: pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go
Comment (slashpai) [inline pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go:877]: missed `omitempty` for optional field

File: pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go
Comment (slashpai) [inline pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go:879]: missed `omitempty` for optional field

File: pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go
Comment (slashpai) [inline pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go:884]: missed `omitempty` for optional field

File: pkg/prometheus/resource_selector.go
Comment (slashpai) [inline pkg/prometheus/resource_selector.go:874]:
```suggestion
			rejectFn(sc, fmt.Errorf("nomadSDConfigs: %w", err))
```

File: pkg/prometheus/resource_selector.go
Comment (slashpai) [inline pkg/prometheus/resource_selector.go:1200]: nit: move after `validateHetznerSDConfigs` just to keep the order of added SD's

File: pkg/prometheus/resource_selector_test.go
Comment (slashpai) [inline pkg/prometheus/resource_selector_test.go:2431]: nit: move to end just to keep the order of added tests and SD's

**Review by slashpai — CHANGES_REQUESTED** [review]
a few nits

**Comment by dongjiang1989** [comment]
> a few nits

Thanks. Fixed.

**Review by slashpai — APPROVED** [review]

**Comment by slashpai** [comment]
@dongjiang1989 I have the merged the changes as CI didn't report any issues but looks like its complaining in generate action now https://github.com/prometheus-operator/prometheus-operator/actions/runs/8799810147/job/24149623425

can you do a follow-up PR?

**Comment by dongjiang1989** [comment]
> @dongjiang1989 I have the merged the changes as CI didn't report any issues but looks like its complaining in generate action now https://github.com/prometheus-operator/prometheus-operator/actions/runs/8799810147/job/24149623425
> 
> can you do a follow-up PR?

Thanks. 👌 I'll add a follow-up PR

**Pull request merged**
